root/edit/details/AddAnnotation.js

```diff
@@ -0,0 +1,67 @@
+/*
+ * @flow
+ * Copyright (C) 2018 Shamroy Pellew
+ *
+ * This file is part of MusicBrainz, the open internet music database,
+ * and is licensed under the GPL version 2, or (at your option) any
+ * later version: http://www.gnu.org/licenses/gpl-2.0.txt
+ */
+
+import React from 'react';
+
+import {addColon, l} from '../../static/scripts/common/i18n';
+import DescriptiveLink from '../../static/scripts/common/components/DescriptiveLink';
+import formatEntityTypeName from '../../static/scripts/common/utility/formatEntityTypeName';
+
+const AddAnnotation = ({edit}: {edit: EditT}) => (
+  <table
+    className={`details add-${edit.display_data.entity_type}-annotation`}
+  >
+    <table
+      className={`details add-${edit.display_data.entity_type}-annotation`}
+    >
+      {edit.display_data[edit.display_data.entity_type] || !edit.preview ? (
```

Comment:
edit.preview is a problem with Flow, but tbh I'm not sure where this is even coming from, so I'm not sure what type to add :/

Reply:
It comes from `Edit::Role::Preview`, thought not all edit types consume this role. So in `Edit::TO_JSON` you'll need a conditional check:

```Perl
my $can_preview = $self->does('MusicBrainz::Server::Edit::Role::Preview');
return {
    # ...
    $can_preview ? (preview => boolean_to_json($self->preview)) : (),
    # ...
};
```

And the Flow type should be `preview?: boolean`.